File: src/LeetCode1TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class LeetCode1TwoSum {
    public static void main(String[] args) {
        int[] ans = twoSum(new int[]{3,3}, 6);
        if (ans != null) {
            System.out.println(ans[0]);
            System.out.println(ans[1]);
        }
    }

    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            if(map.containsKey(target - nums[i])){
                return new int[]{i, map.get(target - nums[i])};
            }
            map.put(nums[i], i);
        }
        return null;
    }
}

File: src/LeetCode1TwoSum.kt
fun main(args: Array<String>) {
    twoSum(intArrayOf(2, 7, 11, 15), 9)
    twoSum(intArrayOf(3, 2, 4), 6)
    twoSum(intArrayOf(3, 3), 6)
}

fun twoSum(nums: IntArray, target: Int): IntArray {
    nums.forEachIndexed { i, _ ->
        for (j in i + 1 until nums.size) {
            if (nums[i] + nums[j] == target) {
                println(i)
                println(j)
                return intArrayOf(i, j)
            }
        }
    }
    return intArrayOf()
}

fun twoSum1(nums: IntArray, target: Int): IntArray {
    val map = HashMap<Int, Int>()
    nums.forEachIndexed { index, i ->
        if (map.containsKey(target - i)) {
            return intArrayOf(index, map.getOrDefault(target - i, -1))
        } else {
            map[i] = index
        }
    }
    return intArrayOf()
}